WordPress/gutenberg
# Block mover button: do not show focus styles on pointer interactions

## What?


Tweak mover button styles in both the block toolbar and the list view, so that the button's focus styles are not shown as the results of a user click.

## Why?


More polished button styles when interacting with a pointer.

## How?


By using the [`:focus-visible` CSS pseudoselector](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible), which lets the browser determine via heuristics if the focus should be made evident on the element — in practice, it allows for focus styles to be shown only on keyboard interaction (and not as a result of pointer clicks).

 - moved the block mover button styles to `packages/block-editor/src/components/block-mover/style.scss`
- I still had to leave those same styles in place in `packages/components/src/toolbar/style.scss` because they are applied to the rest of toolbar button (using `:focus` instead of `:focus-visible`)
- I had to add an override in the block mover button styles to remove the `:focus` styles applied by the block toolbar
- I had to leave some minor overrides for the size/position of the `::before` pseudo element in the list view

_Note: the correct way of making this change would be by changing styles directly in the `Button` component. This PR instead resorts to applying style overrides directly in the block editor — this is a sub-optimal approach, which introduces technical debt and fragmentation, making it actually harder to work on the `Button` component in the future. This approach is therefore not reccommended._

## Testing Instructions





Block mover buttons in the toolbar:
- on desktop viewport sizes:
  1. Select a block
  2. In the toolbar, click on the up/down block mover buttons
  3. Notice how the focus ring doesn't show
  4. Navigate the toolbar with the keyboard (tab & and arrow keys)
  5. Notice how the mover buttons show a focus rings when focus via keyboard interaction
- Resize the viewport to mobile size, the toolbar should move to a fixed positoin on top of the editor
  1. Repeat steps 1 to 5 above

Block mover buttons in the tree list:
- enable block mover buttons (I did it by [setting this `hasRenderedMovers` variable to `true`](https://github.com/WordPress/gutenberg/blob/820d8cfb1366354867cd212a51c9388a971888ad/packages/block-editor/src/components/list-view/block.js#L137))
- Repeat steps 2 to 5 above

## Screenshots or screencast 

### Before (`trunk`): focus ring appears on both pointer and keyboard interactions

Block toolbar:

https://user-images.githubusercontent.com/1083581/196880269-345440ab-16d0-46c4-bd3a-a4edafd5bbd0.mp4

List view:

https://user-images.githubusercontent.com/1083581/196880283-7ce52515-de7c-42b6-b8db-1ccc59c9b369.mp4

### After (this PR): focus ring only shows on keyboard interactions

Block toolbar:

https://user-images.githubusercontent.com/1083581/196880289-7c2d888e-5fcd-4e89-ab10-dd7d4f5a5063.mp4

List view:

https://user-images.githubusercontent.com/1083581/196880300-b7bceb93-cc2b-44c4-b113-904dcfb726b8.mp4
